sdk/src/Core/Amazon.Runtime/Credentials/ImmutableCredentials.cs

```diff
@@ -55,6 +56,14 @@ public class ImmutableCredentials
         /// Account-based endpoints take the form https://accountid.ddb.region.amazonaws.com
         /// </summary>
         public string AccountId { get; private set; }
+
+        /// <summary>
+        /// Gets a cached AWS4a signing key for the current credentials.
+        /// </summary>
+        /// <remarks>
+        /// The key must not be disposed, and must not be returned to the user.
+        /// </remarks>
+        internal ECDsa AWS4aSigningKey { get; set; }
```

Comment:
I had the idea of caching the signing key in the `ImmutableCredentials` instance, but it will not work in practice, because we are cloning the immutable credentials before passing them to the signer ([example](https://github.com/aws/aws-sdk-net/blob/9b002cd7e308fc79e2ed80954dcb707f859c2bf4/sdk/src/Core/Amazon.Runtime/Credentials/BasicAWSCredentials.cs#L76)), which doesn't make sense because the immutable credentials objects are immutable.

What to do? Should I open another PR to remove the copyings?